.github/workflows/lint.yml

```diff
@@ -0,0 +1,68 @@
+name: Lint
+#
+# This workflow lints the UI library and reports back any suggested fixes
+#
+
+on:
+    workflow_dispatch:
+    workflow_call:
+
+permissions:
+    contents: read
+    pull-requests: write
+
+defaults:
+    run:
+        shell: bash
+
+jobs:
+    # --- Lint pre-compiled assets for consistency --- #
+    lint:
+        name: Lint
+        runs-on: ubuntu-latest
+        timeout-minutes: 5
+        outputs:
+            eslint_diff_files: ${{ steps.changed-files.outputs.eslint_diff_files }}
+            stylelint_diff_files: ${{ steps.changed-files.outputs.stylelint_diff_files }}
+        steps:
+            # install but don't build - we're linting pre-compiled assets
+            - name: Check out code
+              uses: actions/checkout@v4
+              with:
+                  fetch-depth: 2
+
+            - name: Setup Job and Install Dependencies
+              uses: ./.github/actions/setup-job
```

Comment:
this includes a build step. we might consider removing it from the setup and make jobs run build explicitly